demos/bookstore/app/models/cart.ts

```diff
@@ -10,27 +12,44 @@ export interface Cart {
   items: CartItem[]
 }
 
-// Store carts by session ID
+// Store carts by cartId, cartId will be stored in the session
+let nextCartId = 1
```

Comment:
No longer storing by session id because that's not something app code can always _directly_ access (`sesson.id` isn't populated on initial session creation - only when we eventually commit the session).  We don't want to commit a session until they interact with it, so there's no good way to key a cart by sessionId the first time we interact with the session.  Instead, we use a cartId that we store in the session (which "activates" the session and triggers the cookie), and we key the cart lookup by that.